packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-table.vue

```diff
@@ -0,0 +1,206 @@
+						@update:model-value="emit('update-value', [slotProps.data.value])"
+					/>
+				</div>
+				<span v-else-if="slotProps.data.type === ParamType.CONSTANT">
+					<InputNumber
+						class="p-inputtext-sm"
+						inputId="numericInput"
+						mode="decimal"
+						:min-fraction-digits="1"
+						:max-fraction-digits="10"
+						v-model.lazy="slotProps.data.value.value"
+						@update:model-value="emit('update-value', [slotProps.data.value])"
+					/>
+				</span>
+			</template>
+		</Column>
+		<Column field="source" header="Source" style="width: 35%"></Column>
+		<Column field="visibility" header="Visibility" style="width: 10%">
+			<template #body="slotProps">
+				<InputSwitch v-model="slotProps.data.visibility" @click.stop />
+			</template>
+		</Column>
+		<template #expansion="slotProps">
+			<tera-model-config-table
+				hide-header
+				v-if="slotProps.data.type === ParamType.MATRIX"
+				:model-configuration="modelConfiguration"
+				:stratified-model-type="stratifiedModelType"
+				:data="slotProps.data.tableFormattedMatrix"
+				:table-type="tableType"
+				@update-value="(val: ModelParameter | Initial) => emit('update-value', [val])"
+			/>
+		</template>
+	</Datatable>
+	<Teleport to="body">
+		<tera-stratified-matrix-modal
+			v-if="matrixModalContext.isOpen && stratifiedModelType"
+			:id="matrixModalContext.matrixId"
+			:model-configuration="modelConfiguration"
+			:stratified-model-type="stratifiedModelType"
+			:stratified-matrix-type="matrixModalContext.stratifiedMatrixType"
+			:open-value-config="matrixModalContext.isOpen"
+			@close-modal="matrixModalContext.isOpen = false"
+			@update-configuration="updateConfigFromMatrix"
+		></tera-stratified-matrix-modal>
+	</Teleport>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+import Button from 'primevue/button';
+import InputNumber from 'primevue/inputnumber';
+import type { ModelConfiguration, ModelParameter, Initial } from '@/types/Types';
+import { StratifiedModel } from '@/model-representation/petrinet/petrinet-service';
+import { StratifiedMatrix } from '@/types/Model';
+import Datatable from 'primevue/datatable';
+import Column from 'primevue/column';
+import TeraStratifiedMatrixModal from '@/components/model/petrinet/model-configurations/tera-stratified-matrix-modal.vue';
+import { ParamType } from '@/types/common';
+import Dropdown from 'primevue/dropdown';
+import InputSwitch from 'primevue/inputswitch';
+import { pythonInstance } from '@/python/PyodideController';
+import InputText from 'primevue/inputtext';
+
+const typeOptions = [
+	{ label: 'Constant', value: ParamType.CONSTANT },
+	{ label: 'Distribution', value: ParamType.DISTRIBUTION },
+	{ label: 'Time Varying', value: ParamType.TIME_SERIES }
+];
+const props = defineProps<{
+	modelConfiguration: ModelConfiguration;
+	data: any[];
+	stratifiedModelType: StratifiedModel | null;
+	tableType: StratifiedMatrix;
+	hideHeader?: boolean;
+}>();
```

Comment:
it feels like we should be able to have some notion of typing for `data` at this stage? Since you created the tableFormattedXYZ right beforehand?

Reply:
I've got away with removing both tableType and stratifiedModelType.  tableType was a little tricky, I needed to add an additional emitter to the component instead called 'update-config' so that the parent can can the updates.  stratifiedModelType was extracted from the model-configuration as we discussed